trieste/models/keras/models.py

```diff
@@ -249,7 +250,10 @@ def predict(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
         :return: The predicted mean and variance of the observations at the specified
             ``query_points``.
         """
-        ensemble_distributions = self.ensemble_distributions(query_points)
+        # handle leading batch dimensions
+        input_dims = len(self.model.input_shape[0])
```

Comment:
Just for my understanding, you look at `input_shape[0]` (instead of the whole shape) because that is the input shape of the first network in the ensemble (and we assume all networks have the same inputs shape); is that correct?

Reply:
(though now that the docs gym issue is fixed, there's an error in the deep ensemble notebook that suggest that I might be not quite right; will investigate...)

Reply:
Turns out that keras's `Functional` class has code to "allow (None,) and (None, 1) Tensors to be passed interchangeably", which the notebook was making use of. I've changed my fix so it doesn't break that (slightly odd) usage.